src/TeamStore.Keeper/Services/AssetService.cs

```diff
@@ -193,6 +193,38 @@ IApplicationIdentityService applicationIdentityService
             return retrievedAssets;
         }
 
+        // searching assets should preload all assets
+        public async Task<List<AssetSearchResult>> GetAssetResultsAsync()
+        {
+            // Validate current user
+            var currentUser = await _applicationIdentityService.GetCurrentUser();
+            if (currentUser == null) throw new Exception("Unauthorised requests are not allowed."); ;
+
+            // Get assets with appropriate access checks
+            var retrievedAssets = await _dbContext.Assets.Where(a =>
+                a.IsArchived == false &&
+                a.Project.AccessIdentifiers.Any(ai => ai.Identity.Id == currentUser.Id))
+                .Include(p => p.Project.AccessIdentifiers)
+                .ThenInclude(p => p.Identity) // NOTE: intellisense doesn't work here (23.09.2017) https://github.com/dotnet/roslyn/issues/8237
+                .ToListAsync();
+
+            // LOG access asset - open project? TODO
+            var assetResults = new List<AssetSearchResult>(retrievedAssets.Count);
+            foreach (var asset in retrievedAssets)
+            {
+                DecryptAsset(asset);
+                assetResults.Add(
+                    new AssetSearchResult()
```

Comment:
Here the service performs mappings, which should happen in the Factory mapping layer like the rest of the solution. Consistency is a critical ingredient for maintainable software.